### PR TITLE
Add more PersonServiceTest-#902

### PR DESCRIPTION
**Pull request by Agaba-derrick**
# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the Issue number if applicable.
- [ ] The PR includes a video showing the changes for the work done.
- [x] The PR title follows conventional commit label standards.
- [x] The changes confirm to the OpenElis Global x3 Styleguide and design documentation.
- [x] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing Guidelines of this project.

## Summary
Added at least 90% of the PersonService Test
## Screenshots
Screenshot from 2024-06-11 17-37-41

## Related Issue

 Add more integration tests to the Patient and Person service #902 

## Other
@mozzy11 kindly Requesting for Review here

### CODE REVIEW

File: src/test/java/org/openelisglobal/person/PersonServiceTest.java [hunk 1]
```diff
@@ -45,4 +48,151 @@ private Person createPerson(String firstName, String LastName) {
...
+        String firstName = "John";
+        String lastName = "Doe";
+        Person person = createPerson(firstName, lastName);
+        person.setCity("New York");
+        person.setCountry("USA");
+        person.setState("NY");
+        person.setStreetAddress("123 Main St");
+        person.setZipCode("10001");
+
+        System.out.println("Creating person with details: " + person);
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/person/PersonServiceTest.java:90]: we dont need these

**Review by mherman22 — CHANGES_REQUESTED** [review]
Thanks @Agaba-derrick  for this, it generally LGTM,

please add tests for https://github.com/mherman22/OpenELIS-Global-2/blob/564de765f77329a9c3d54c588d03f2a47130ca2c/src/main/java/org/openelisglobal/person/service/PersonServiceImpl.java#L59-L69 aswell

File: src/main/java/org/openelisglobal/person/daoimpl/PersonDAOImpl.java [hunk 2]
```diff
@@ -142,16 +142,15 @@ public Person getPersonByLastName(String lastName) throws LIMSRuntimeException {
     @Override
     @Transactional(readOnly = true)
     public Person getPersonById(String personId) throws LIMSRuntimeException {
-        String sql = "From Person p where id = :personId";
+        String sql = "From Person p where p.id = :personId";
         try {
             Query<Person> query = entityManager.unwrap(Session.class).createQuery(sql, Person.class);
-            query.setParameter("personId", Integer.parseInt(personId));
```
Comment (mozzy11) [inline src/main/java/org/openelisglobal/person/daoimpl/PersonDAOImpl.java:148]: why this change ??

**Comment by Agaba-derrick** [comment]
> Thanks @Agaba-derrick for this, it generally LGTM,
> 
> please add tests for https://github.com/mherman22/OpenELIS-Global-2/blob/564de765f77329a9c3d54c588d03f2a47130ca2c/src/main/java/org/openelisglobal/person/service/PersonServiceImpl.java#L59-L69 aswell

added

**Pull request closed without merging**